subprojects/packagefiles/lzo/lzotest/meson.build

```
@@ -2,8 +2,9 @@ t = executable('lzotest', 'lzotest.c', dependencies: lzo2_dep)
 
 test('basic lzotest', t, args: ['-mlzo', '-n2', '-q', files('../COPYING')], suite: 'basic')
 test('exhaustive lzotest', t, args: ['-mavail', '-n10', '-q', files('../COPYING')])
-if build_machine.system() != 'windows'
-    test('lzotest directory check', find_program('../util/check.sh'),
+check = find_program('../util/check.sh', required: false)
+if check.found()
+    test('lzotest directory check', check,
       args: [topdir],
```

```suggestion
shell = find_program('sh', required: false)
if shell.found()
    test('lzotest directory check', shell,
      args: [topdir / 'util/check.sh', topdir],
```

This is what I meant.

---

It's not clear to me why that is especially advantageous, since on unix the sh shell always exists, and on Windows it probably doesn't but Meson will anyways manually parse the shebang, extract /bin/sh, and search for /bin/sh as the "real" program.

---

Interesting, why is meson trying to run this test on windows for now then?

---

Because you can install the "sh" program on Windows, and the search algorithm for /bin/sh -- a POSIX path -- on Windows is to remove the POSIX path and search for "sh".

---

But actually trying to run that file fails with code 127, which is AFAIK exactly because `/bin/sh` is not found by OS. I guess implementing change I've suggested would fix this, but it also raises a question if path stripping should be done in this case - meson reports program as found yet it can't be executed in practice, which is undesirable.